typescript/src/electrum.ts

```diff
@@ -120,19 +138,51 @@ export class Client implements BitcoinClient {
    * @returns Promise holding the outcome.
    */
   private async withElectrum<T>(action: Action<T>): Promise<T> {
-    const electrum = new Electrum(
-      this.credentials.host,
-      this.credentials.port,
-      this.credentials.protocol,
-      this.options
-    )
+    const connect = async (credentials: Credentials): Promise<Electrum> => {
+      const electrum: Electrum = new Electrum(
+        credentials.host,
+        credentials.port,
+        credentials.protocol,
+        this.options
+      )
 
-    try {
       await this.withBackoffRetrier()(async () => {
-        return await electrum.connect("tbtc-v2", "1.4.2")
+        // FIXME: Connection timeout should be a property of the Electrum client.
+        // Since it's not configurable in `electrum-client-js` we add timeout
+        // as a workaround here.
+        return pTimeout(
+          (async () => {
+            try {
+              await electrum.connect("tbtc-v2", "1.4.2")
+              await electrum.server_ping()
+              return
+            } catch (error) {
+              throw new Error(`Electrum server connection failure: [${error}]`)
+            }
+          })(),
+          this.connectionTimeout,
+          `timed out on electrum connect after ${this.connectionTimeout} ms`
+        )
       })
-    } catch (error) {
-      throw new Error(`Electrum server connection failure: [${error}]`)
+
+      return electrum
+    }
+
+    let electrum: Electrum | undefined = undefined
+    for (const credentials of this.credentials) {
+      try {
+        electrum = await connect(credentials)
+      } catch (err) {
+        console.warn(
+          `failed to connect to electrum server: [${credentials.protocol}://${credentials.host}:${credentials.port}]: ${err}`
+        )
+        continue
+      }
+      break
+    }
```

Comment:
Would a loop like this be a bit simpler to read?
```
    for (const credentials of this.credentials) {
      try {
        electrum = await connect(credentials)
        break
      } catch (err) {
        console.warn(
          `failed to connect to electrum server: [${credentials.protocol}://${credentials.host}:${credentials.port}]: ${err}`
        )
      }
    }
```

There would only be a `break` without `continue`.

Reply:
Thanks for the suggestion! Done in e3056d8627a887693d69104b501983861d831453